src/pages/todolist.tsx

```diff
@@ -0,0 +1,139 @@
+import Head from 'next/head';
+import { useState } from 'react';
+import styled from 'styled-components';
+
+import { colors } from '../constants/theme';
+
+interface ListItem {
+  ind: number;
+  val: string;
+  setList: Function;
+}
+
+const ListItem = ({ ind, val, setList }: ListItem) => {
+  const [done, setDone] = useState(false);
+
+  const handleRemove = (e: any) => {
+    e.preventDefault();
+    setList((old: any) => {
+      old.splice(ind, 1);
+      return [...old];
+    });
+  };
+
+  return (
+    <li
+      key={ind + val}
+      onClick={() => setDone(done => !done)}
+      onContextMenu={handleRemove}
+      style={done ? { textDecoration: 'line-through', color: 'rgba(0, 0, 0, 0.2)' } : {}}
+    >
+      {val}
+    </li>
+  );
+};
```

Comment:
Do not add helper functions/ type declared/ styled-components at the top. Keep the main component at the top.